### ISSUE TITLE
Parent joint transforms not getting applied

### ISSUE THREAD

**Issue by aaronsnoswell** [robcog-iai/URoboSim-depr#24]
I have a URDF for the [Kinova MOVO robot](http://www.kinovarobotics.com/innovation-robotics/movobeta/) that I'm trying to import into Unreal engine.

The URDF is valid (renders in Rviz correctly - see below), however doesn't import into UE4 correctly.

![image](https://user-images.githubusercontent.com/967771/39094242-d49cf508-466f-11e8-991c-b11a38133d50.png)

When I import it using this plugin, I've noticed that after about 3 levels of joint-link hierarchy, the joints no longer have the correct transforms (see UE4 screenshot, below). After some extensive investigation it seems that under some (unknown) circumstances, joints get created, but without their parent transforms first being applied. If I manually update the URDF for the culprit joints, applying the transforms in the parent hierarchy of joints, then that part of the model will import correctly.

![image](https://user-images.githubusercontent.com/967771/39094303-b9b9a7ee-4670-11e8-9ca0-d5996889d212.png)

In the above screenshot the arms and head have imported correctly, but are inside the torso, because they don't have their parent transforms applied.

**Comment by andreihaidu** [comment]
Hi @aaronsnoswell we noticed multiple bugs in the current state, that is why we started doing it from scratch with the lessons learned.

Also URDFs will no longer be supported since one can describe robots that cannot be physically simulated. (e.g. virtual links attached to real joints etc). That is why SDF (from gazebo) will be used for the robot description. There is however a urdf to sdf tool supported by them. Once the state is in a usable one, we will release the code. Right now it is in a very early, half working one.

So about the bug, we will probably not have time to look over the issue. And if it is not in a hurry for you, I would wait for the new version. I ma hoping it will get in a usable state in 3-4 weeks.

Cheers!
